components/camera/fake/image_file_test.go

```diff
@@ -68,19 +64,17 @@ func TestColor(t *testing.T) {
 	cam, err := newCamera(ctx, resource.Name{API: camera.API}, cfg, logger)
 	test.That(t, err, test.ShouldBeNil)
 
-	stream, err := cam.Stream(ctx)
-	test.That(t, err, test.ShouldBeNil)
-
 	_, err = cam.NextPointCloud(ctx)
 	test.That(t, err, test.ShouldNotBeNil)
 
 	readInImage, err := rimage.NewImageFromFile(artifact.MustPath("vision/objectdetection/detection_test.jpg"))
 	test.That(t, err, test.ShouldBeNil)
 
-	strmImg, _, err := stream.Next(ctx)
+	imgBytes, _, err := cam.Image(ctx, utils.MimeTypeJPEG, nil)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, strmImg, test.ShouldResemble, readInImage)
-	test.That(t, strmImg.Bounds(), test.ShouldResemble, readInImage.Bounds())
+	expectedBytes, err := rimage.EncodeImage(ctx, readInImage, utils.MimeTypeJPEG)
```

Comment:
why are you encoding the bytes? Can't you just read the `artifact.MustPath("vision/objectdetection/detection_test.jpg")` file & compare the bytes?

Reply:
Unfortunately, the bytes compared this way are not equal. The way the test is set up right now emulates what image_file.go is doing: 

1. [Read in the file and turn it into an rimage.Image](https://github.com/viamrobotics/rdk/blob/d298b6b95592838b1e0ea96d16f2c879340d62fc/components/camera/fake/image_file.go#L101)
2. Then videosourcewrapper's Image wrapper around ReadImage would then encode it using rimage https://github.com/hexbabe/sean-rdk/blob/65b5bad742bf59c60e249d74f22338c06753465f/components/camera/videosourcewrappers.go#L209

Reply:
Gotcha, I didn't realize that the image_file camera processed the image before returning it. 

In my opinion, in the long term, there should be an option for the image_file camera to just return the bytes of the image without changing them, but that doesn't need to be done as part of this change. 

This is fine for this pr.

Thanks for explaining.

Reply:
followup ticket https://viam.atlassian.net/browse/RSDK-9280